### PR TITLE
[core] Fix of incorrect resizing of TileCache

### LINKED ISSUE #15926: [core] Assert check failed in TileCache::add
**Platform:** linux and macOS

**Mapbox SDK version:** checkout from master 1d49c23b8558e8c24a51e7ba84525ddd06f231be

I'm using sdk for parallel generate some thousand static map images and sometimes i have failed assert check in tiles cache.
https://github.com/mapbox/mapbox-gl-native/blob/4c1d03c257754ecc1555f3e31b3c6ccf45c4bf7b/src/mbgl/tile/tile_cache.cpp#L37

Workaround for me was correcting condition 
https://github.com/mapbox/mapbox-gl-native/blob/4c1d03c257754ecc1555f3e31b3c6ccf45c4bf7b/src/mbgl/tile/tile_cache.cpp#L24

```c++
if (!tiles.emplace(key, std::move(tile)).second) {
```

Please check this code.

### PR DESCRIPTION
**Pull request by syershov**
In some case resizing TileCache we can give a assert check failed.
```Assertion failed: (orderedKeys.size() <= size), function add, file /Users/darkserj/sandbox/mapbox-gl-native/src/mbgl/tile/tile_cache.cpp, line 38.```

Fixes are remove all mentions the key from orderedKeys in case resizing and left only single mention the key at end of orderedKeys in case adding.

Fixes #15926

### CODE REVIEW

File: src/mbgl/tile/tile_cache.cpp [hunk 1]
```diff
@@ -8,7 +8,7 @@ void TileCache::setSize(size_t size_) {
 
     while (orderedKeys.size() > size) {
         auto key = orderedKeys.front();
-        orderedKeys.pop_front();
+        orderedKeys.remove(key);
```
Comment (pozdnyakov) [inline src/mbgl/tile/tile_cache.cpp]: why is this change?

Comment (syershov) [reply src/mbgl/tile/tile_cache.cpp]: In the case of reducing the size of the cache, a situation could turn out when the tile was no longer there and the mention of it in the orderedKeys remained, which led to the triggering of the assert.
This was possible due to a second error in the if-condition with emplace.

**Comment by syershov** [comment]
@pozdnyakov Hi, what i need to do for merge PR?

**Comment by pozdnyakov** [comment]
@syershov Thanks for your contribution! I've just made a CP PR for this change so that our CI can pass + added some minor indentation fixes on top.

**Pull request closed without merging**